### PR TITLE
[query] Add support for dropping NaNs when rendering results

### PR DESCRIPTION
**Pull request by benraskin92**

### CODE REVIEW

File: src/cmd/services/m3query/config/config.go [hunk 1]
```diff
@@ -128,6 +128,10 @@ type Configuration struct {
 
 	// Cache configurations.
 	Cache CacheConfiguration `yaml:"cache"`
+
+	// DropNaNs drops NaNs before returning query results.
+	// If you want to simulate Promtheus exactly, set this to true.
+	DropNaNs bool `yaml:"dropNaNs"`
```
Comment (arnikola) [inline src/cmd/services/m3query/config/config.go]: Don't have this as a top level config setting

File: src/query/api/v1/handler/prometheus/native/common_test.go [hunk 2]
```diff
@@ -212,11 +218,109 @@ func TestRenderResultsJSON(t *testing.T) {
...
+	valsWithNaN.SetValueAt(1, math.NaN())
+
+	series := []*ts.Series{
+		ts.NewSeries([]byte("foo"),
+			valsWithNaN, test.TagSliceToTags([]models.Tag{
+				models.Tag{Name: []byte("bar"), Value: []byte("baz")},
+				models.Tag{Name: []byte("qux"), Value: []byte("qaz")},
+			})),
+		ts.NewSeries([]byte("bar"),
+			ts.NewFixedStepValues(10*time.Second, 2, 2, start), test.TagSliceToTags([]models.Tag{
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/native/common_test.go]: Have one of these tests have some nans

File: src/cmd/services/m3query/config/config.go (hunk 1 above)
Comment (benraskin92) [reply src/cmd/services/m3query/config/config.go]: What do you propose? Maybe have like a `QueryOptions` struct?

Comment (arnikola) [reply src/cmd/services/m3query/config/config.go]: `Result` would make more sense imo; also should default to true

Comment (arnikola) [reply src/cmd/services/m3query/config/config.go]: Maybe rename it to `returnNansInQueryResult` or something

File: src/query/api/v1/handler/prometheus/native/common_test.go (hunk 2 above)
Comment (benraskin92) [reply src/query/api/v1/handler/prometheus/native/common_test.go]: Actually `TestRenderResultsJSONWithDroppedNaNs` already has a `NaN` and a real number

File: src/cmd/services/m3query/config/config.go [hunk 3]
```diff
@@ -155,6 +156,22 @@ type FilterConfiguration struct {
...
 
+// ResultOptions are the result options for query.
+type ResultOptions struct {
+	// DropNaNsInResults drops NaNs before returning query results.
+	// The default is true, which matches Prometheus
+	DropNaNsInResults *bool `yaml:"dropNaNs"`
+}
+
+// DropNaNsOrDefault returns the provided input or default if none is provided
+func (c ResultOptions) DropNaNsOrDefault() *bool {
```
Comment (arnikola) [inline src/cmd/services/m3query/config/config.go]: This should probably return a `bool` not a `*bool`

Comment (benraskin92) [reply src/cmd/services/m3query/config/config.go]: Shoot yeah meant to change that.

File: src/cmd/services/m3query/config/config.go (hunk 1 above)
Comment (andrewmains12) [reply src/cmd/services/m3query/config/config.go]: +1 on `ResultOptions` (or `Result`)

File: src/cmd/services/m3query/config/config.go [hunk 4]
```diff
@@ -146,6 +151,22 @@ type FilterConfiguration struct {
 	CompleteTags Filter `yaml:"completeTags"`
 }
 
+// ResultOptions are the result options for query.
+type ResultOptions struct {
+	// DropNaNsInResults drops NaNs before returning query results.
+	// The default is true, which matches Prometheus
+	DropNaNsInResults *bool `yaml:"dropNaNs"`
```
Comment (andrewmains12) [inline src/cmd/services/m3query/config/config.go]: Nit: if `DropNan's` is the default, could rename to `KeepNaN's`, and use the builtin default for bools? Avoids the pointer.

Comment (arnikola) [inline src/cmd/services/m3query/config/config.go]: Also can you call it `dropNans` instead, the NaN is annoying and out of place in a config

**Pull request merged**